### PR TITLE
Adding validation against policy field

### PR DESCRIPTION
**Pull request by rjzak**

### CODE REVIEW

File: src/ext/snp/mod.rs [hunk 1]
```diff
@@ -24,31 +24,84 @@ pub struct Evidence<'a> {
...
+        /// Indicates if only one socket is permitted
+        SingleSocket = 0b00010000u8,
+        /// Indicates if debugging is permitted
+        Debug = 0b00001000u8,
+        /// Indicates is association with migration assistant is permitted
+        MigrateMA = 0b00000100u8,
+        /// Reserved, must be one
+        Reserved = 0b00000010u8,
+        /// Indicates if SMT is permitted
+        SMT = 0b00000001u8,
```
Comment (npmccallum) [inline src/ext/snp/mod.rs:38]:
```suggestion
        SMT = 1 << 0,
```

File: src/ext/snp/mod.rs [hunk 2]
```diff
@@ -24,31 +24,84 @@ pub struct Evidence<'a> {
...
+flags! {
+    pub enum Flags: u8 {
+        /// Indicates if only one socket is permitted
+        SingleSocket = 0b00010000u8,
+        /// Indicates if debugging is permitted
+        Debug = 0b00001000u8,
+        /// Indicates is association with migration assistant is permitted
+        MigrateMA = 0b00000100u8,
+        /// Reserved, must be one
+        Reserved = 0b00000010u8,
```
Comment (npmccallum) [inline src/ext/snp/mod.rs:36]:
```suggestion
        Reserved = 1 << 1,
```

File: src/ext/snp/mod.rs [hunk 3]
```diff
@@ -24,31 +24,84 @@ pub struct Evidence<'a> {
     pub report: &'a [u8],
 }
 
+flags! {
+    pub enum Flags: u8 {
+        /// Indicates if only one socket is permitted
+        SingleSocket = 0b00010000u8,
```
Comment (npmccallum) [inline src/ext/snp/mod.rs:30]:
```suggestion
        SingleSocket = 1 << 4,
```

This is more idiomatic and easier to read. It says that `SingleSocket is the fifth bit` (i.e. 4, but zero-indexed).

File: src/ext/snp/mod.rs [hunk 4]
```diff
@@ -24,31 +24,84 @@ pub struct Evidence<'a> {
...
+
+#[repr(C, packed)]
+#[derive(Copy, Clone, Debug)]
+pub struct Policy {
+    /// Minimum ABI minor version required for the guest to run
+    pub abi_minor: u8,
+    /// Minimum ABI major version required for the guest to run
+    pub abi_major: u8,
+    /// Bit fields indicating enabled features
+    pub bit_fields: FlagSet<Flags>,
```
Comment (npmccallum) [inline src/ext/snp/mod.rs:50]:
```suggestion
    pub flags: FlagSet<Flags>,
```

File: src/ext/snp/mod.rs [hunk 5]
```diff
@@ -24,31 +24,84 @@ pub struct Evidence<'a> {
...
+    /// Reserved, must be zero
+    rsvd2: u8,
+    /// Reserved, must be zero
+    rsvd1: u32,
```
Comment (npmccallum) [inline src/ext/snp/mod.rs:54]: Just make this `[u8; 5]` as this is easier to reason about.

**Review by npmccallum — CHANGES_REQUESTED** [review]

**Comment by npmccallum** [comment]
@rjzak Also, your life will be much happier if you configure your editor to run `cargo fmt` on every save.

File: src/ext/snp/mod.rs [hunk 6]
```diff
@@ -4,18 +4,18 @@ use std::{fmt::Debug, mem::size_of};
...
 use der::{Decodable, Encodable, Sequence};
+use flagset::{flags, FlagSet};
 use pkcs8::AlgorithmIdentifier;
 use sha2::Digest;
 use x509::ext::Extension;
 use x509::{request::CertReqInfo, Certificate};
 use x509::{PkiPath, TbsCertificate};
 
-use super::ExtVerifier;
-
```
Comment (npmccallum) [inline src/ext/snp/mod.rs:18]: Don't move this.

**Review by npmccallum — CHANGES_REQUESTED** [review]

File: src/ext/snp/mod.rs (hunk 6 above)
Comment (rjzak) [reply src/ext/snp/mod.rs]: I think my IDE re-ordered it for some reason.

**Comment by rjzak** [comment]
I addressed the changes but Github seems to think they're unresolved.

**Review by npmccallum — APPROVED** [review]

**Pull request merged**